app/models/experimentation/notifications_experiment.rb

```diff
@@ -50,12 +50,26 @@ def enroll_patients(date, limit = MAX_PATIENTS_PER_DAY)
         .each_record { |patient| random_treatment_group.enroll(patient, reporting_data(patient, date)) }
     end
 
-    def monitor(date)
-      # Add check_notification_statuses in a follow up PR
+    def monitor
+      report_notification_statuses
       mark_visits
       evict_patients
     end
 
+    def report_notification_statuses
+      treatment_group_memberships
+        .joins(treatment_group: :reminder_templates)
+        .where("messages -> reminder_templates.message ->> 'status' = ?", :pending)
+        .select("messages -> reminder_templates.message ->> 'notification_id' AS notification_id")
+        .select("reminder_templates.message, treatment_group_memberships.*")
+        .in_batches(of: MEMBERSHIPS_BATCH_SIZE).each_record do |membership|
```

Comment:
Should we add a note to take a look at the performance of this query later, and consider adding appropriate json indexes?